spec/integration/network/http/api/indirected_routes_spec.rb

```diff
@@ -52,5 +54,36 @@
         end
       end
     end
+
+    describe "an error from IndirectedRoutes", :if => Puppet.features.rack? do
+      let(:handler) { Puppet::Network::HTTP::RackREST.new }
+
+      describe "returns json" do
+        it "when a standard error" do
+          response = Rack::Response.new
+          request = Rack::Request.new(
+            Rack::MockRequest.env_for("/puppet/v3/invalid-indirector"))
+
+          handler.process(request, response)
+
+          expect(response.header["Content-Type"]).to match(/json/)
+          resp = JSON.parse(response.body.first)
+          expect(resp["message"]).to match(/The indirection name must be purely alphanumeric/)
+          expect(resp["issue_kind"]).to be_a_kind_of(String)
+        end
+        it "when a server error" do
+          response = Rack::Response.new
+          request = Rack::Request.new(
+            Rack::MockRequest.env_for("/puppet/v3/unknown_indirector"))
+
+          handler.process(request, response)
+
+          expect(response.header["Content-Type"]).to match(/json/)
+          resp = JSON.parse(response.body.first)
+          expect(resp["message"]).to match(/Could not find indirection/)
+          expect(resp["issue_kind"]).to be_a_kind_of(String)
```

Comment:
👍 on this test, exactly what I *expected* (pun intended) to find, based on the ticket description.